### PR TITLE
[SYCL] Release notes for May'20 DPCPP implementation update

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

File: sycl/ReleaseNotes.md [hunk 1]
```diff
@@ -1,3 +1,166 @@
...
+    [ABI policy guide](doc/ABIPolicyGuide.md)
+  - Using `cl::sycl::program` API to refer to a kernel defined in another
+    translation unit leads to undefined behavior
+  - Linkage errors with the following message:
+    `error LNK2005: "bool const std::_Is_integral<bool>" (??$_Is_integral@_N@std@@3_NB) already defined`
+    can happen when a SYCL application is built using MS Visual Studio 2019
+    version below 16.3.0
+    The workaround is to enable `-std=c++17` for the failing MSVC version.
+
+## Prerequisites
```
Comment (bader) [inline sycl/ReleaseNotes.md:150]: I think we should point to exact revision of https://github.com/intel/llvm/blob/sycl/buildbot/dependency.conf used for testing 67d3d9e.

Comment (pvchupin) [reply sycl/ReleaseNotes.md:150]: I think we should probably just remove prerequisites from release notes, but we need GetStartedGuide.md update first. 
Basically idea is that dependency.conf for corresponding tag or any commit should always point to the recommended dependencies for this commit.
I also don't want to block May'20 release notes by this cleanup needed.

Comment (pvchupin) [reply sycl/ReleaseNotes.md:150]: #1699 is in progress, so let's remove prerequisites from here.

Comment (romanovvlad) [reply sycl/ReleaseNotes.md]: Removed.

File: sycl/ReleaseNotes.md [hunk 2]
```diff
@@ -1,3 +1,154 @@
...
+    [ABI policy guide](doc/ABIPolicyGuide.md)
+  - Using `cl::sycl::program` API to refer to a kernel defined in another
+    translation unit leads to undefined behavior
+  - Linkage errors with the following message:
+    `error LNK2005: "bool const std::_Is_integral<bool>" (??$_Is_integral@_N@std@@3_NB) already defined`
+    can happen when a SYCL application is built using MS Visual Studio 2019
+    version below 16.3.0
+    The workaround is to enable `-std=c++17` for the failing MSVC version.
+
+Please, see the runtime installation guide [here](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#install-low-level-runtime)
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:150]: Please remove this line as well.

Comment (romanovvlad) [reply sycl/ReleaseNotes.md]: Done.

**Comment by romanovvlad** [comment]
Added new known issue while the patch is still on review.

**Pull request merged**